### DISCUSSION TITLE
Tiempo Parte 1 test hard

### DISCUSSION THREAD

**Discussion by antoniablanco** [IIC2133-PUC/2024-1#143] [Tarea 2] [answered]
Hola, estaba revisando el test hard 2, y al multiplicar (N = 46550 ~ 4 * 10^4; Q = 32822 ~ 3 * 10^4) N* Q = 1.52786*10^9 = 15 * 10^8 ~ 15 segundos. Lo cual es muy justo con respecto a los 10 segundos pedidos en el enunciado. Esto sin contar el tiempo de lectura, escritura de archivos y el cálculo del módulo en el rolling hashing.

**Comment by benjavicente** [comment, marked as answer]
Hola! La complejidad es **proporcional** al tiempo dado el tamaño del input. Eso quiere decir que **NO** puedes calcular el tiempo basándote solo en el tamaño de los inputs, necesitarías una referencia par realizar una estimación. 

Los test hards de la parte 1 se pueden pasar con un tiempo menor a 10s, en la complejidad pedida. Es importante que cuando se habla de complejidad, hay un factor constante que se ignora. Entender ese factor es importante para los test hard (y no es necesario optimizar la lectura y escritura de archivos para bajarlo significativamente).

**Reply by leonelcespedess** [reply]
Mientras se cumpla con la complejidad, no es problema el tiempo de los hard?

**Reply by benjavicente** [reply]
Si es problema. Tenemos que ponerle un límite al tiempo de ejecución para no quedar indefinidamente corriendo tests.

En la parte, ha pasar que el factor constante es alto, dado que se repiten operaciones cada vez al hacer rolling del hash en vez de calcularla 1 vez  y usarla luego.
